### PR TITLE
Add requesting user id parameter to key claim methods in `TransportLayerClient`

### PR DESCRIPTION
**Pull request by H-Shay**

### CODE REVIEW

File: synapse/federation/federation_client.py [hunk 1]
```diff
@@ -238,13 +238,15 @@ async def claim_client_keys(
         self,
         destination: str,
         query: Dict[str, Dict[str, Dict[str, int]]],
+        requester: UserID,
```
Comment (clokep) [inline synapse/federation/federation_client.py:241]: Can we use `user_id` or `user` and not `requester`? We almost exclusively use that to mean a `Requester` instance.

**Comment by clokep** [comment]
(I didn't review this, just had a drive-by comment.)

File: synapse/handlers/e2e_keys.py [hunk 2]
```diff
@@ -698,12 +699,12 @@ async def claim_one_time_keys(
         failures: Dict[str, JsonDict] = {}
 
         @trace
-        async def claim_client_keys(destination: str) -> None:
+        async def claim_client_keys(destination: str, user: UserID) -> None:
```
Comment (clokep) [inline synapse/handlers/e2e_keys.py:702]: I don't think you need to pass this in (like we don't pass in `remote_queries`)? It doesn't make much difference though.

File: synapse/federation/federation_client.py [hunk 3]
```diff
@@ -238,13 +238,15 @@ async def claim_client_keys(
...
         query: Dict[str, Dict[str, Dict[str, int]]],
+        user: UserID,
         timeout: Optional[int],
     ) -> JsonDict:
         """Claims one-time keys for a device hosted on a remote server.
 
         Args:
             destination: Domain name of the remote homeserver
             content: The query content.
+            user: The user id of the requesting user
```
Comment (clokep) [inline synapse/federation/federation_client.py:249]: I think we usually put the `user` as one of the early arguments (my brain really wants it to be `user`, `destination`, `query`).

It doesn't really matter though and we can leave it, but I'm curious if there's a reason you chose to put it in the location it is in?

Comment (H-Shay) [reply synapse/federation/federation_client.py:249]: I'm happy to change it if that's the general pattern - I didn't have much reason to put it where it is other than I generally default to adding new parameters to the end of a parameter list.

Comment (clokep) [reply synapse/federation/federation_client.py]: It is a required parameter anyway, so I don't think adding it at the end matters much. (And this isn't even at the end!)

**Review by clokep — APPROVED** [review]
Should be OK if CI is happy.

File: synapse/federation/federation_client.py (hunk 3 above)
Comment (H-Shay) [reply synapse/federation/federation_client.py]: Sorry I meant end of the required parameters - but thank you for pointing this out, this is the sort of convention I am trying to internalize so I appreciate knowing about it :)

**Comment by H-Shay** [comment]
Merged #15663 into develop, thanks for the review!

**Pull request merged**